setup.py

```diff
@@ -37,21 +37,26 @@ def package_files(directory):
         'Flask',
         'numpy',
         'pandas',
-        'scipy',
-        'scikit-learn',
         'python-dateutil',
         'protobuf>=3.6.0',
         'gitpython>=2.1.0',
         'pyyaml',
-        'boto3>=1.7.12',
         'querystring_parser',
         'simplejson',
-        'mleap>=0.8.1',
         'docker>=3.6.0',
         'entrypoints',
         'sqlparse',
         'sqlalchemy',
     ],
+    extras_require={
+        'alldeps':[
```

Comment:
`extras`

Reply:
I agree that 'extras' rather than 'alldeps' might be a better name, if that's what this comment indicates.

Would it also make sense to include other optional dependencies (e.g. tensorflow) in this?

Reply:
Good question. There may be reasons to avoid including tensorflow as part of the pip dependencies -- for example, if people are recommended to [install it using conda](https://towardsdatascience.com/stop-installing-tensorflow-using-pip-for-performance-sake-5854f9d9eb0c). It's also 400 MB by itself, which is the same as the rest of `mlflow[extras]` combined.

I'm not opposed to putting it here, but maybe we can consider it OK to add more `extras` in the future, and start with "this is enough to run a server against common backends and run basic examples in the repo (which often use scikit-learn)"